src/coreclr/tools/SuperFileCheck/Program.cs

```
@@ -0,0 +1,623 @@
+        const string CommandLineCheckPrefixes = "--check-prefixes";
+        const string CommandLineCheckPrefixesEqual = "--check-prefixes=";
+        const string CommandLineInputFile = "--input-file";
+        const string SyntaxDirectiveFullLine = "-FULL-LINE:";
+        const string SyntaxDirectiveFullLineNext = "-FULL-LINE-NEXT:";
+
+        static string FileCheckPath;
+
+        static Program()
+        {
+            // Determine the location of LLVM FileCheck as being next to
+            // the location of SuperFileCheck
+            var superFileCheckPath = typeof(Program).Assembly.Location;
+            if (String.IsNullOrEmpty(superFileCheckPath))
+            {
+                throw new Exception("Invalid SuperFileCheck path.");
+            }
+            var superFileCheckDir = Path.GetDirectoryName(superFileCheckPath);
+            if (superFileCheckDir != null)
+            {
+                FileCheckPath = Path.Combine(superFileCheckDir, "FileCheck");
+            }
+            else
+            {
+                FileCheckPath = "FileCheck";
+            }
+        }
+
+        /// <summary>
+        /// Checks if the given string contains LLVM "<prefix>" directives, such as "<prefix>:", "<prefix>-LABEL:", etc..
+        /// </summary>
+        static bool ContainsCheckPrefixes(string str, string[] checkPrefixes)
+        {
+            // LABEL, NOT, SAME, etc. are from LLVM FileCheck https://llvm.org/docs/CommandGuide/FileCheck.html
+
+            // FULL-LINE and FULL-LINE-NEXT are not part of LLVM FileCheck - they are new syntax directives for SuperFileCheck to be able to
+            // match a single full-line, similar to that of LLVM FileCheck's --match-full-lines option.
+
+            var pattern = $"({String.Join('|', checkPrefixes)})+?({{LITERAL}})?(:|-LABEL:|-NOT:|-SAME:|-EMPTY:|-COUNT:|-DAG:|{SyntaxDirectiveFullLine}|{SyntaxDirectiveFullLineNext})";
```

This made me look at how `{LITERAL}` works.  Perhaps `FULL-LINE` should also be a `<modifier>` for consistency?  It could be an error for cases where it doesn't make sense, but then the parsing would be a bit simpler.  (And we wouldn't silently ignore `CHECK-FULL-LINE-LABEL`, for example.)

---

I think we should do this if we plan on adding more of these new directives/modifiers - at the moment we just have two.

---

Let's discuss this one some more.  I keep coming back to this and wonder if some things would be clearer if handled the same way as filecheck.
- I got stuck thinking SyntaxDirectiveFullLineNext would never get hit (FullLine is a prefix of FullLineNext) before finally realizing that both have the colon on the end (therefore it is -not- a prefix)>
- Does it matter that CHECK-NEXT is missing?
- Why does it use `+?` after the prefixes?  Isn't that `*`, but more importantly why can there be more than one?

---

> Does it matter that CHECK-NEXT is missing?
Looks like I forgot to add that.

> Why does it use +? after the prefixes? Isn't that *, but more importantly why can there be more than one?
I think it doesn't matter if there is more than one, all we want to check is if there is one. For using `+?`, that is a good question - I guess we could use `*`?

> I got stuck thinking SyntaxDirectiveFullLineNext would never get hit (FullLine is a prefix of FullLineNext) before finally realizing that both have the colon on the end (therefore it is -not- a prefix)>
The way I've been thinking of this is `<prefix><directive>` where `<directive>` is `-<directive name>:`..